### PR TITLE
[WIP] Filebrowser 1.0.1

### PR DESCRIPTION
**Pull request by lemavri**
TODO:

- [ ] Send response to different end-point than `data`
- [x] Add fileretrieval action that uploads the specified file to an end-point

### CODE REVIEW

File: lib/agent/providers.js [hunk 1]
```diff
@@ -65,10 +65,22 @@ var map = exports.map = function(cb){
...
 
-var get = exports.get = function(name, cb) {
-
-  var args = arguments,
-      callback = cb || function(){ /* noop */ };
+var get = exports.get = function(name, extra) {
+
+  var cb,
+      options,
+      args = arguments,
```
Comment (raliste) [inline lib/agent/providers.js]: I would suggest, for consistency and styling, to either order the variables alphabetically or put value variables at the beginning (or at the end).

File: lib/agent/providers/files/tree.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        type: 'folder',
+        children: results
+      });
+
+    list.forEach(function(file) {
+      file = path.resolve(dir, file);
+      fs.stat(file, function(err, stat) {
+        if (stat && stat.isDirectory()) {
+          directoryTreeToObj(file, function(err, res) {
+						var dirname = path.dirname(file),
```
Comment (raliste) [inline lib/agent/providers/files/tree.js]: misalignment?

File: lib/agent/providers/files/tree.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+              name: name,
+              path: p,
+              isFile: stat.isFile(),
+              children: res
+            });
+            if (!--pending)
+              done(null, results);
+          });
+        } else {
+					var dirname = path.dirname(file),
```
Comment (raliste) [inline lib/agent/providers/files/tree.js]: idem

File: lib/agent/providers/users/index.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+"use strict";
+
+//////////////////////////////////////////
+// (c) 2011 - Fork Ltd.
```
Comment (raliste) [inline lib/agent/providers/users/index.js]: As new file update the copyright mark to Prey 2015

File: lib/agent/providers/users/index.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+// By Mauricio Schneider - http://preyhq.com
```
Comment (raliste) [inline lib/agent/providers/users/index.js]: Use public domain preyproject.com

File: lib/agent/providers/users/mac.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+"use strict";
+
+////////////////////////////////////////////
+// (c) 2011 - Fork Ltd.
```
Comment (raliste) [inline lib/agent/providers/users/mac.js]: idem

File: lib/agent/providers/files/tree.js (hunk 2 above)
Comment (lemavri) [inline lib/agent/providers/files/tree.js]: Indeed. Vim got funny with the tabs.

**Pull request merged**